google-cloud-spanner/src/test/java/com/google/cloud/spanner/SessionPoolOptionsTest.java

```diff
@@ -186,4 +186,19 @@ public void setNegativeIdleTimeThreshold() {
     SessionPoolOptions.newBuilder()
         .setInactiveTransactionRemovalOptions(inactiveTransactionRemovalOptions);
   }
+
+  @Test
+  public void setAcquireSessionTimeout() {
+    SessionPoolOptions sessionPoolOptions =
```

Comment:
We should add some checks for the corner case timeout values and also verify that we can get it as milliseconds (or that it fails in the correct way):
1. `null` timeout
2. `Duration.ofMillis(0)`
3. `Duration.ofSeconds(Long.MAX_VALUE)`

Reply:
* Sure there was an existing code path that was testing a similar behaviour. I have used that as a reference to add a new test `blockAndTimeoutOnPoolExhaustion_withAcquireSessionTimeout`. 
* For the existing test `blockAndTimeoutOnPoolExhaustion`, I had to explicitly set `acquireSessionTimeout` value as null since otherwise it was going into the new code path and thus failing. This kind of demonstrates how this change is a breaking change and how the customers can mitigate by setting it to `null`.